domain/agentbinary/service/service_test.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package service
+
+import (
+	"context"
+
+	"github.com/juju/testing"
+	jc "github.com/juju/testing/checkers"
+	"go.uber.org/mock/gomock"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/domain/agentbinary"
+)
+
+type serviceSuite struct {
+	testing.IsolationSuite
+
+	mockModelState, mockControllerState *MockAgentBinaryState
+}
+
+var _ = gc.Suite(&serviceSuite{})
+
+func (s *serviceSuite) setupMocks(c *gc.C) *gomock.Controller {
+	ctrl := gomock.NewController(c)
+	s.mockModelState = NewMockAgentBinaryState(ctrl)
+	s.mockControllerState = NewMockAgentBinaryState(ctrl)
+	return ctrl
+}
+
+func (s *serviceSuite) TestListAgentBinaries(c *gc.C) {
```

Comment:
If you feel like it I like to see the contract called out that we are trying to test. I think this case we are asserting that models bins always take first seat.